src/Catel.Extensions.DynamicObjects/Catel.Extensions.DynamicObjects.Shared/Data/DynamicObservableObjectMetaObject.cs

```diff
@@ -42,6 +43,15 @@ public DynamicObservableObjectMetaObject(Expression parameter, DynamicObservable
         }
 
         /// <summary>
+        /// Returns the enumeration of all dynamic member names.
+        /// </summary>
+        /// <returns>The list of dynamic member names.</returns>
+        public override IEnumerable<string> GetDynamicMemberNames()
+        {
+            return ((DynamicObservableObject)Value).GetPropertyNames();
```

Comment:
I'm not sure in 100%, but rather no. Value is passed as parameter before in DynamicObservableObject.GetMetaObject() { return new DynamicObservableObjectMetaObject(parameter, this); } So observable object is value parameter in DynamicObservableObjectMetaObject ctor and is forwarded down to base class. Not sure what happens there, but I think they will not nullify this value.